lib/utils.ts

```diff
@@ -90,6 +90,25 @@ export const Utils = new class {
 		} else {
 			constructor = 'null';
 		}
+
+		// if it has a toString, check that to see if it tells us if there's a base class, else look at the constructor name
+		// (this is to pull the base class so it works for subclasses)
+		let baseName = value?.toString ? /\[object (.*)\]/.exec(value.toString())?.[1] : null;
+		if (!baseName) baseName = constructor;
+
+		switch (baseName) {
+		// other constructor cases can be handled here!
```

Comment:
```suggestion
```
Did you forget about this? This is now rule C1 in CONTRIBUTING:

https://github.com/smogon/pokemon-showdown/blob/master/CONTRIBUTING.md#c1-dont-teach-javascript